### PR TITLE
chore(ts-migration): migrates clear refinements connector, widgets an…

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request migrates the clear refinements connector, widgets, and stories content. Note: 

There is a couple of questions, that's why I have putted the **draft** state.

### CODE REVIEW

File: src/connectors/clear-refinements/connectClearRefinements.ts [hunk 1]
```diff
@@ -7,73 +7,83 @@ import {
...
- *
- * // mount widget on the page
- * search.addWidgets([
- *   customClearRefinementsWidget({
- *     containerNode: $('#custom-clear-all-container'),
- *   })
- * ]);
- */
-export default function connectClearRefinements(renderFn, unmountFn = noop) {
+export type ClearRefinementsConnectorParamsItem = {
```
Comment (nunomaduro) [inline src/connectors/clear-refinements/connectClearRefinements.ts:17]: I asked on slack, but I am quite sure that an item on clear refinements is a simple string.

File: src/connectors/clear-refinements/connectClearRefinements.ts [hunk 2]
```diff
@@ -7,73 +7,83 @@ import {
...
+   * Function to transform the items passed to the templates.
+   */
+  transformItems?: TransformItems<ClearRefinementsConnectorParamsItem>;
+};
+
+export type ClearRefinementsRendererOptions = {
+  /**
+   * Triggers the clear of all the currently refined values.
+   */
+  refine: () => void;
```
Comment (nunomaduro) [inline src/connectors/clear-refinements/connectClearRefinements.ts:57]: Can you confirm that there is no param here?

File: src/connectors/clear-refinements/connectClearRefinements.ts [hunk 3]
```diff
@@ -7,73 +7,83 @@ import {
...
+
+  /**
+   * Indicates if search state is refined.
+   */
+  hasRefinements: boolean;
+
+  /**
+   * Creates a url for the next state when refinements are cleared.
+   */
+  createURL: CreateURL<void>;
```
Comment (nunomaduro) [inline src/connectors/clear-refinements/connectClearRefinements.ts:67]: Can you confirm that there is no param here?

File: src/connectors/clear-refinements/connectClearRefinements.ts (hunk 1 above)
Comment (nunomaduro) [reply src/connectors/clear-refinements/connectClearRefinements.ts:17]: On the test for example, there are simple strings: https://github.com/algolia/instantsearch.js/pull/4385/files#diff-93bf43d5d7e3988ea2803945111d3d80L558.

Comment (nunomaduro) [reply src/connectors/clear-refinements/connectClearRefinements.ts:17]: And same on docs: https://www.algolia.com/doc/api-reference/widgets/clear-refinements/js/#widget-param-transformitems.

**Comment by Haroenv** [comment]
In the future we can probably avoid this connector and use currentRefinements instead, like in React

File: src/connectors/clear-refinements/connectClearRefinements.ts (hunk 1 above)
Comment (francoischalifour) [reply src/connectors/clear-refinements/connectClearRefinements.ts:17]: You're right, they're strings. In these cases, you can [try it in a sandbox](https://codesandbox.io/s/instantsearchjs-clear-refinements-qebvy?file=/src/app.js).

File: src/connectors/clear-refinements/connectClearRefinements.ts (hunk 2 above)
Comment (francoischalifour) [reply src/connectors/clear-refinements/connectClearRefinements.ts:57]: Yep

https://github.com/algolia/instantsearch.js/blob/bae6ed2875b03a8a43c576085a19f2ceb43ba1f2/src/connectors/clear-refinements/connectClearRefinements.js#L130-L141

File: src/connectors/clear-refinements/connectClearRefinements.ts (hunk 3 above)
Comment (francoischalifour) [reply src/connectors/clear-refinements/connectClearRefinements.ts:67]: Yep

https://github.com/algolia/instantsearch.js/blob/bae6ed2875b03a8a43c576085a19f2ceb43ba1f2/src/connectors/clear-refinements/connectClearRefinements.js#L143-L153

**Pull request merged**